doc/designs/prci_checker.md

```diff
@@ -0,0 +1,149 @@
+[File] nightly_previous [Jobs] 149
+[File] nightly_latest_pki [Jobs] 75
+[File] gating [Jobs] 24
+[File] nightly_rawhide [Jobs] 149
+[File] nightly_latest_389ds [Jobs] 45
+[File] nightly_latest [Jobs] 149
+[File] nightly_latest_selinux [Jobs] 149
+[File] nightly_latest_testing [Jobs] 149
+[File] nightly_latest_sssd [Jobs] 16
+[File] nightly_latest_testing_selinux [Jobs] 149
+CHECKS FINISHED SUCCESSFULLY
+```
+
+**alternatively:**
+
+``` bash
+$ make yamllint
+[...]
+Check PRCI definitions
+-----------
+/usr/bin/python ./ipatests/prci_definitions/prci_checker.py -d ./ipatests/prci_definitions -s ./ipatests/prci_definitions/prci_jobs_spec.yaml;
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] nightly_previous [Jobs] 149
+[...]
+[File] nightly_latest_testing_selinux [Jobs] 149
+CHECKS FINISHED SUCCESSFULLY
+-----------
+```
+
+**To check a specific file:**
+
+``` bash
+$ python3 ipatests/prci_definitions/prci_checker.py -f ipatests/prci_definitions/gating.yaml
+BEGINNING PRCI JOB DEFINITIONS CHECKS
+[File] gating [Jobs] 24
+CHECKS FINISHED SUCCESSFULLY
+```
+
+## Tool specification file
+
+Most part of the checks carried out by the script are described in the specification file [prci_jobs_spec.yaml](https://github.com/freeipa/freeipa/blob/master/ipatests/prci_definitions/prci_jobs_spec.yaml).
+
+The yaml file contains suppported classes, definitions that should have the same number of jobs, and format specification for each file.
+
+**File content can vary according to the different needs in every branch**. [Example for `ipa-4-10`](https://github.com/freeipa/freeipa/blob/ipa-4-10/ipatests/prci_definitions/prci_jobs_spec.yaml):
+
+``` yaml
+# Specification file for PRCI definitions used by prci_checker script
+
+# List of supported test-run classes for non-build jobs
+classes:
+  - "RunPytest"
+  - "RunPytest2"
+  - "RunPytest3"
+  - "RunWebuiTests"
+  - "RunADTests"
+
+# (Optional) Definition files that should contain the same number of jobs
+fixed_n_jobs:
+  - nightly_ipa-4-10_latest
+  - nightly_ipa-4-10_latest_selinux
+  - nightly_ipa-4-10_previous
+
+# Info specific to prci definition files
+# 'job_prefix' field is mandatory
+# 'update_packages', 'selinux_enforcing', 'enable_testing_repo' and
+# 'copr_defined' are optional boolean fields (if not specified,
+# false value is assumed).
+# New definitions specifications may be added anytime
+prci_job_spec:
+  gating:
+    job_prefix: 'fedora-latest-ipa-4-10/'
+  nightly_ipa-4-10_latest:
+    job_prefix: 'fedora-latest-ipa-4-10/'
+  nightly_ipa-4-10_latest_selinux:
+    job_prefix: 'fedora-latest-ipa-4-10/'
+    selinux_enforcing: True
+  nightly_ipa-4-10_previous:
+    job_prefix: 'fedora-previous-ipa-4-10/'
+```
+
+## Error cases examples
+
+After adding new jobs to the definition or editing the content, it is common that a developer has made some mistakes following the convention. This checker will help us correcting these errors. Examples:
+
+-   Wrong or unsupported class:
```

Comment:
Should this be a sub-heading instead of a bullet? e.g. ### Wrong or unsupported class

Reply:
Yep, changed